src/java/htsjdk/samtools/util/SequenceUtil.java

```diff
@@ -848,7 +922,10 @@ public static String calculateMD5String(final byte[] data, final int offset, fin
     }
 
     /**
-     * A rip off samtools bam_md.c
+     * Calculate MD and NM similarly to Samtools, except that N->N is a match.
+     *
+     * TODO this calculates NM differently than calculateSamNmTag
+     * TODO (specifically, it does not check set relationships between ambiguous bases.
```

Comment:
It looks like this method was originally added as part of the CRAM changes. I'm not sure what will be affected by changing it, so I leave it for now and open a separate ticket.